source/material_model/utilities.cc

```diff
@@ -1151,7 +1151,7 @@ namespace aspect
         // Retrieve the list of composition names
         const std::vector<std::string> list_of_composition_names = this->introspection().get_composition_names();
 
-        n_phase_transitions_per_composition.reset(new std::vector<unsigned int>());
+        n_phase_transitions_per_composition = std::make_unique<std::vector<unsigned int>>();
```

Comment:
Just out of curiosity, is it better to do an assignment than to reset?

Reply:
It makes no practical difference, but the C++ community has really tried to avoid using bare `new` operator because it is easy to introduce memory leak bugs when you forget the corresponding `delete` operator. As a consequence, all style guides these days suggest to use `std::make_shared` or `std::make_unique` where possible, avoid `new` and `delete`.

There is no risk here because we pass the newly created object to a class that manages its lifetime (the `std::unique_ptr` object), but I chose to just follow the usual style.

Reply:
Thanks for the explanation!